garage/tf/models/base.py

```diff
@@ -0,0 +1,313 @@
+"""Base model classes."""
+import abc
+import warnings
+
+import tensorflow as tf
+
+
+class BaseModel(abc.ABC):
+    """Interface-only abstract class for models."""
+
+    def build(self, *inputs):
+        """
+        Output of model with the given input placeholder(s).
+
+        This function is implemented by subclasses to create their computation
+        graphs, which will be managed by Model. Generally, subclasses should
+        implement `build()` directly.
+        """
+        pass
+
+    @property
+    def name(self):
+        """Name for this Model."""
+        pass
+
+    @property
+    def input(self):
+        """Tensor input of the Model."""
+        pass
+
+    @property
+    def output(self):
+        """Tensor output of the Model."""
+        pass
+
+    @property
+    def inputs(self):
+        """Tensor inputs of the Model."""
+        pass
+
+    @property
+    def outputs(self):
+        """Tensor outputs of the Model."""
+        pass
+
+    @property
+    def parameters(self):
+        """Parameters of the Model."""
+        pass
+
+    @parameters.setter
+    def parameters(self, parameters):
+        """Set parameters of the Model."""
+        pass
+
+
+class AutoPickableModel(BaseModel, metaclass=abc.ABCMeta):
```

Comment:
i don't think this needs to be called autopickleable anymore. it can just be `Model`.

we may want to merge this with the `TfModel` class if we think that we won't have non-TF models in the future.

Reply:
In that case, `Model` by default will handle pickle and if someone wants to have another pickle implementation, he will inherit from `BaseModel`?

Reply:
yes.